src/commands/extract-inline-scripts.yml

```diff
@@ -0,0 +1,93 @@
+description: >
+  Extract shell script snippets from `run` blocks within the packed
+  source  code (single file) of an orb and store each snippet as a
+  temporary file to be checked by the shellcheck orb. Requires Bash
+  v4.0 or later.
+
+parameters:
+  target-filepath:
+    type: string
+    default: orb.yml
+    description: >
+      Where is the YAML file containing orb source code? Defaults to a
+      file named `orb.yml` in the working directory
+
+  output-dir:
+    type: string
+    default: scripts
+    description: >
+      Where should scripts be outputted? Defaults to a directory named
+      `scripts` (it will be created if it does not exist) in the working
+      directory
+
+  run-pack:
+    type: boolean
+    default: true
+    description: >
+      Pack a destructured orb source directory for this orb command to use?
+
+  pack-source:
+    type: string
+    default: src
+    description: Path to the root folder of an unpacked orb
+
+  pack-destination:
+    type: string
+    default: orb.yml
+    description: Path including the filename to output the packed orb
+
+  debug:
+    type: boolean
+    default: false
+    description: >
+      Extra output for orb developers
+
+steps:
+  - build-tools/install-ci-tools
+
+  - circleci-cli/install
+
+  - when:
+      condition: <<parameters.run-pack>>
+      steps:
+        - orb-tools/pack:
+            source: <<parameters.pack-source>>
+            destination: <<parameters.pack-destination>>
+
+  - run:
+      name: Extract shell scripts
+      shell: /bin/bash
+      command: |
+        readarray ORBSOURCE < <<parameters.target-filepath>>
+
+        i=0
+        until [ $i -eq ${#ORBSOURCE[@]} ]; do
+          LINE="${ORBSOURCE[$i]}"
+          echo "$LINE"
+          if echo "$LINE" | grep "command: |"; then
```

Comment:
This is not a YAML parser, and won't be able to extract strings from YAML files correctly. You need to use a YAML parser here, and extract any string values located at the `jobs: <job-name>: steps: <step> run: command: <script>` .

From the docs, ShellCheck is also only documented to check `bash` and `sh`, so you will need to filter for those in the `shell` key - otherwise this will run shellcheck over scriptes written in `powershell`, `cmd`, `zsh`, etc.

Reply:
true i suppose it'd be much easier to use something like `yq` lol